### PR TITLE
fix(sqllab): clean comments within quotes

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY
minor regression from #23378

The regex for commented out block also clean up the block within the quotes.
Therefore following query threw a syntax error since it removes the quoted value.

```
SELECT * FROM table WHERE column = '--unknown--';
==> 
SELECT * FROM table WHERE column = '
```

This commit adds the condition to skip the quoted block.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
After:

```
SELECT * FROM table WHERE column = '--unknown--';
==> 
SELECT * FROM table WHERE column = '--unknown--';
```

Before:

```
SELECT * FROM table WHERE column = '--unknown--';
==> 
SELECT * FROM table WHERE column = '
```

### TESTING INSTRUCTIONS
Added tests for edge cases

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by michael-s-molina — APPROVED** [review]
LGTM but it would be nice to get @villebro and @betodealmeida approvals too.

File: superset-frontend/src/SqlLab/actions/sqlLab.js [hunk 1]
```diff
@@ -357,12 +357,70 @@ export function fetchQueryResults(query, displayLimit) {
...
+  const chunks = splitByQuotedBlock(sql);
+  return (
+    chunks
+      // replace quoted blocks in a hash format
+      .map((chunk, index) =>
+        quotes.includes(chunk[0]) ? `${quotedBlockHash}:${index}:` : chunk,
+      )
+      .join('')
+      // Clean out the commented-out blocks
+      .replace(/(--.*?$|\/\*[\s\S]*?\*\/)\n?/gm, '\n')
```
Comment (ktmud) [inline superset-frontend/src/SqlLab/actions/sqlLab.js:416]:
```suggestion
      .replace(/(--.*$|\/\*[\s\S]*\*\/)\n?/gm, '\n')
```

I don't think these question marks are necessary

**Review by ktmud — APPROVED** [review]
Interesting approach

**Comment by villebro** [comment]
@justinpark @michael-s-molina @ktmud we have a fair deal of logic for this in the backend - is there some reason we want to parse for this kind of stuff in the frontend? It feels like we should centralize this in one place.

**Comment by ktmud** [comment]
I agree this logic is probably better to live in the backend---it's more useful to record the original SQL users wrote, too.

Not sure how comments are removed in the backend but my guess is it happens after we render the Jinja template (which is why we saw the error in https://github.com/apache/superset/pull/23378 in the first place).

Maybe we can change it to remove comments before rendering, but how would you know the comments are safe to remove without rendering the template? Also, if we are removing comments using a SQL parser, the parser is likely to fail before you render the template. Not sure what's the best way forward, but since we are already remove comments in the frontend, it's probably still worth merging this PR first.

**Comment by justinpark** [comment]
> Maybe we can change it to remove comments before rendering, but how would you know the comments are safe to remove without rendering the template? Also, if we are removing comments using a SQL parser, the parser is likely to fail before you render the template. Not sure what's the best way forward, but since we are already removing comments in the frontend, it's probably still worth merging this PR first.

I agree with [ktmud](https://github.com/ktmud)'s idea

**Review by villebro — APPROVED** [review]
@ktmud @justinpark I agree, this is an improvement to the status quo; we can follow up later by moving this into the backend. @michael-s-molina you can check `superset/sql_parse.py` which we use for parsing queries in the backend, and I seem to recall having added tests that ensure `'--comment'` aren't stripped out, which is very similar to what's being checked here.

**Comment by justinpark** [comment]
> I agree, this is an improvement to the status quo; we can follow up later by moving this into the backend.

Sounds good. yeah definitely backend is a better place. until then, we can use this solution

File: superset-frontend/src/SqlLab/actions/sqlLab.js (hunk 1 above)
Comment (justinpark) [reply superset-frontend/src/SqlLab/actions/sqlLab.js:416]: This `?` mark covers the multiple comment group case

i.e.
```
/*
comments
*/
no-comments
/*
comments
*/
```

**Pull request merged**